src/codegen/llvm/codegen_llvm_helper_visitor.cpp

```diff
@@ -534,17 +529,43 @@ void CodegenLLVMHelperVisitor::visit_nrn_state_block(ast::NrnStateBlock& node) {
     /// convert local statement to codegenvar statement
     convert_local_statement(*loop_block);
 
-    /// create for loop node
-    auto for_loop_statement = std::make_shared<ast::CodegenForStatement>(initialization,
-                                                                         condition,
-                                                                         increment,
-                                                                         loop_block);
+    /// main loop possibly vectorized on vector_width
+    {
+        /// loop constructs : initialization, condition and increment
+        const auto& initialization = create_statement_as_expression("id = 0");
+        const auto& condition = create_expression("id < node_count");
+        const auto& increment = create_statement_as_expression("id = id + {}"_format(vector_width));
+
+        auto for_loop_statement_main = std::make_shared<ast::CodegenForStatement>(initialization,
+                                                                                  condition,
+                                                                                  increment,
+                                                                                  loop_block);
+
+        /// convert all variables inside loop body to instance variables
+        convert_to_instance_variable(*for_loop_statement_main, loop_index_var);
+
+        /// loop itself becomes one of the statement in the function
+        function_statements.push_back(for_loop_statement_main);
+    }
+
+    /// remainder loop possibly vectorized on vector_width
```

Comment:
```suggestion
    /// remainder loop (serial)
```

Reply:
what does it mean? remainder for serial?

Reply:
Here serial means non-SIMD : Let's say vector width is 8 and loop trip count is 10. The remainder loop has trip count is 2 and it's executed in serial fashion / non-simd.